File: Solutions/_1207.cs
public class Solution
{
    public bool UniqueOccurrences(int[] arr)
    {
        Hashtable hash = new Hashtable();
        HashSet<int> hashset = new HashSet<int>();

        for (int i = 0; i < arr.Length; i++)
        {

            hash[arr[i]] = 1 + (int)((hash.ContainsKey(arr[i])) ? hash[arr[i]] : 0);
        }

        foreach (DictionaryEntry item in hash)
        {
            if (!hashset.Add((int)item.Value))
            {
                return false;
            }
        }

        return true;
    }
}

File: Solutions/_1207.go
func uniqueOccurrences(arr []int) bool {
	hashmap := make(map[int]int)
	count := make(map[int]int)

	for _, val := range arr {
		hashmap[val]++
	}

	for _, val := range hashmap {
		if _, ok := count[val]; ok {
			return false
		}
		count[val]++
	}
	return true
}